src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -70,12 +70,16 @@ Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
       if (keep_executing) {
         // Note that the dependencies in the task spec are mutated to only contain
         // plasma dependencies after ResolveDependencies finishes.
-        const SchedulingKey scheduling_key(task_spec.GetSchedulingClass(),
-                                           task_spec.GetDependencyIds(),
-                                           task_spec.IsActorCreationTask()
-                                               ? task_spec.ActorCreationId()
-                                               : ActorID::Nil(),
-                                           task_spec.GetRuntimeEnvHash());
+        const SchedulingKey scheduling_key(
+            task_spec.GetSchedulingClass(),
+            task_spec.GetDependencyIds(),
+            task_spec.IsActorCreationTask() ? task_spec.ActorCreationId()
+                                            : ActorID::Nil(),
+            task_spec.GetRuntimeEnvHash(),
+            task_spec.IsNodeSchedulingStrategy()
+                ? std::make_pair(task_spec.GetNodeSchedulingStrategyNodeId(),
+                                 task_spec.GetNodeSchedulingStrategySoft())
+                : std::make_pair(NodeID::Nil(), false));
```

Comment:
Can probably just say Nil == is not node scheduling strategy. That would allow you to remove `IsNodeSchedulingStrategy` too (or make it alias for checking node_id == Nil).